canonical-web-and-design/maas-ui
# Speed up checkbox checking

## Done
- Wrap machine table columns in React.memo so they don't rerender if the props don't change.
- Wrap the menu change function in useCallback so it doesn't update and force the columns to rerender.
- Remove the menu change function from columns that don't use it.

These changes made the speed of checking all the checkboxes go from ~270ms to ~40ms.

### Before

<img width="409" alt="before checkbox optimise" src="https://user-images.githubusercontent.com/361637/81253802-c9849b00-906c-11ea-803e-542faa68d481.png">

### After

<img width="409" alt="after checkbox optimise" src="https://user-images.githubusercontent.com/361637/81253810-d1dcd600-906c-11ea-89bc-0106088a6300.png">


## QA
- View the machine list.
- Click on a machine, group or all checkbox. The checkbox(es) should get checked quickly.

## Fixes
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/1044.
